speech/google/cloud/speech/streaming_response.py

```diff
@@ -0,0 +1,152 @@
+                   results=results, result_index=result_index)
+
+    @property
+    def confidence(self):
+        """Confidence score for recognized speech.
+
+        :rtype: float
+        :returns: Confidence score of recognized speech [0.0-1.0].
+        """
+        if self.results and self.results[0].alternatives:
+            top_alternative = self.results[0].alternatives[0]
+            return top_alternative.confidence
+        else:
+            return 0.0
+
+    @property
+    def endpointer_type(self):
+        """Endpointer indicating the state of the speech detection.
+
+        :rtype: str
+        :returns: String derived from :class:`~endpointer_type.EndpointerType`.
+        """
+        return self._endpointer_type
+
+    @property
+    def is_final(self):
+        """Represents an interim result that may change.
+
+        :rtype: bool
+        :returns: True if the result has completed it's processing.
+        """
+        if self.results:
+            return self.results[0].is_final
+        else:
+            return False
+
+    @property
+    def result_index(self):
+        """Result index associated with this response.
+
+        :rtype: int
+        :returns: Result index of this response.
+        """
+        return self._result_index
+
+    @property
+    def results(self):
+        """List of results for this response.
+
+        :rtype: list of :class:`~result.StreamingSpeechResult`
+        :returns: List of ``StreamingSpeechResult`` in this response.
+        """
+        return self._results
+
+    @property
+    def transcript(self):
+        """Get most likely transcript from response.
+
+        :rtype: str
+        :returns: Transcript text from response.
+        """
+        if self.results and self.results[0].alternatives:
+            top_alternative = self.results[0].alternatives[0]
+            return top_alternative.transcript
+        else:
+            return None
+
+
+class EndpointerType(StreamingRecognizeResponse.EndpointerType):
+    """Endpointer type for tracking state of Speech API detection.
+
+    ENDPOINTER_EVENT_UNSPECIFIED (int): No endpointer event specified.
+    START_OF_SPEECH (int): Speech has been detected in the audio stream.
+    END_OF_SPEECH (int): Speech has ceased to be detected in the audio
+    stream.
+    END_OF_AUDIO (int): The end of the audio stream has been reached. and
+    it is being processed.
+    END_OF_UTTERANCE (int): This event is only sent when
+    ``single_utterance`` is ``true``. It indicates that the server has
+    detected the end of the user's speech utterance and expects no
+    additional speech. Therefore, the server will not process additional
+    audio. The client should stop sending additional audio data.
+
+    See:
+    https://cloud.google.com/speech/reference/rpc/\
+    google.cloud.speech.v1beta1#endpointertype
+    """
+    REVERSE_MAP = {v: k for k, v
+                   in vars(StreamingRecognizeResponse.EndpointerType).items()}
```

Comment:
I am :-1: to adding this `REVERSE_MAP` **but** it's a missing feature in GAX / GAPIC, methinks, so maybe file an issue? As for this `REVERSE_MAP`.

1.  You can get the reverse map without making the subclass. Ditch the subclass.
1.  Use `StreamingRecognizeResponse.EndpointerType.__dict__` instead of `vars(StreamingRecognizeResponse.EndpointerType)`
1.  No one-char variable names ever (I know it's clear here, but habits and stuff)
1.  Put the `REVERSE_MAP` as a private class constant (right after the imports)

    ```python
    _REVERSE_MAP = {
        value: key for key, value
        in StreamingRecognizeResponse.EndpointerType__dict__.items()
        if not key.startswith('__')}
    ```